core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalWriteConverterCtx.scala

```diff
@@ -17,81 +17,131 @@
+  override def put(colIdx: Int, data: Char): Unit = {
+    chunkWriter.writeChar(data)
+    increaseCounters()
+  }
 
-  override def put(colIdx: Int, data: Short) = externalFrameWriter.sendShort(data)
+  override def put(colIdx: Int, data: Short): Unit = {
+    chunkWriter.writeShort(data)
+    increaseCounters()
+  }
 
-  override def put(colIdx: Int, data: Int) = externalFrameWriter.sendInt(data)
+  override def put(colIdx: Int, data: Int): Unit = {
+    chunkWriter.writeInt(data)
+    increaseCounters()
+  }
 
-  override def put(colIdx: Int, data: Long) = externalFrameWriter.sendLong(data)
+  override def put(colIdx: Int, data: Long): Unit = {
+    chunkWriter.writeLong(data)
+    increaseCounters()
+  }
 
-  override def put(colIdx: Int, data: Float) = externalFrameWriter.sendFloat(data)
+  override def put(colIdx: Int, data: Float): Unit = {
+    chunkWriter.writeFloat(data)
+    increaseCounters()
+  }
 
-  override def put(colIdx: Int, data: Double) = externalFrameWriter.sendDouble(data)
+  override def put(colIdx: Int, data: Double): Unit = {
+    chunkWriter.writeDouble(data)
+    increaseCounters()
+  }
 
-  override def put(colIdx: Int, data: java.sql.Timestamp) = externalFrameWriter.sendTimestamp(data)
+  override def put(colIdx: Int, data: java.sql.Timestamp): Unit = {
+    chunkWriter.writeTimestamp(data)
+    increaseCounters()
+  }
 
   // Here we should call externalFrameWriter.sendDate - however such method does not exist
   // Hence going through the same path as sending Timestamp long value.
-  override def put(colIdx: Int, data: java.sql.Date) = externalFrameWriter.sendLong(data.getTime)
-
-  override def put(colIdx: Int, data: String) = externalFrameWriter.sendString(data)
+  override def put(colIdx: Int, data: java.sql.Date): Unit = {
+    chunkWriter.writeLong(data.getTime)
+    increaseCounters()
+  }
 
-  override def putNA(columnNum: Int) = externalFrameWriter.sendNA()
+  override def put(colIdx: Int, data: String): Unit = {
+    chunkWriter.writeString(data)
+    increaseCounters()
+  }
 
-  override def numOfRows(): Int = externalFrameWriter.getNumberOfWrittenRows
+  override def putNA(columnNum: Int): Unit = {
+    chunkWriter.writeNA(expectedTypes(columnNum))
+    increaseCounters()
+  }
 
   override def putSparseVector(startIdx: Int, vector: SparseVector, maxVecSize: Int): Unit = {
-    externalFrameWriter.sendSparseVector(vector.indices, vector.values)
+    chunkWriter.writeSparseVector(vector.indices, vector.values)
+    increaseCounters()
   }
 
   override def putDenseVector(startIdx: Int, vector: DenseVector, maxVecSize: Int): Unit = {
-    externalFrameWriter.sendDenseVector(vector.values)
+    chunkWriter.writeDenseVector(vector.values)
+    increaseCounters()
   }
 
+  override def numOfRows(): Int = currentRowIdx
+
   override def startRow(rowIdx: Int): Unit = {}
 
   override def finishRow(): Unit = {}
-}
 
+  private def increaseCounters(): Unit = {
```

Comment:
We can remove `increaseCounters` from the code. I think we do not need it anymore as we upload the data via streams. This was originally used on the backend to see when H2O should stop listenting for the incomming data.

Reply:
We can edit this method https://github.com/h2oai/sparkling-water/blob/ad063d605e976d6770730c41024170dcae73a0e7/core/src/main/scala/org/apache/spark/h2o/converters/WriteConverterCtx.scala#L33 to accept dataSize as `Int` and pass there size of the partition instead in both, internal & external cluster case

Reply:
Ok, will do.